### ISSUE TITLE
Is it possible to add a module to accommodate and adjust for the noise in fixed camera poses？

### ISSUE THREAD

**Issue by han-xiangyu** [colmap/colmap#2819]
Hi,

When I use the fix pose sparse reconstruction in autonomous driving datasets such as nuscenes, I always cannot get the good sparse reconstruction bacause of the inaccurate given poses. So I would like to ask if it is possible to have some error tolerance for the fix pose sparse reconstruction to allow the camera to adjust their poses a little bit in order to obtain a better sparse points cloud?

Thanks

**Comment by ahojnnes** [comment]
By creating a sparse reconstruction, I assume you mean (1) populating an existing reconstruction with the poses, (2) running colmap feature extraction + matching, (3) running the point triangulator?

**Comment by tsattler** [comment]
This will not solve the issue. Rather, after the point triangulator, you want to run a few iterations of: (4) refine the poses, and (5) complete and merge tracks. I opened a PR for this a while ago, but with outdated code and I never got around to fixing it.

**Comment by han-xiangyu** [comment]
> By creating a sparse reconstruction, I assume you mean (1) populating an existing reconstruction with the poses, (2) running colmap feature extraction + matching, (3) running the point triangulator?

Yes, that's what I mean. I'm referring to the steps outlined here: https://colmap.github.io/faq.html#reconstruct-sparse-dense-model-from-known-camera-poses

**Comment by han-xiangyu** [comment]
> This will not solve the issue. Rather, after the point triangulator, you want to run a few iterations of: (4) refine the poses, and (5) complete and merge tracks. I opened a PR for this a while ago, but with outdated code and I never got around to fixing it.

Do you mean PR #2145? It seems to be exactly what I’m looking for, and it appears to be in demand by some others as well.

**Comment by ahojnnes** [comment]
This question has been addressed in #2821. Use `pose_prior_mapper` with per-frame covariance (low for known poses, high for uncertain ones), or use `Mapper.fix_existing_images=1` with an initial reconstruction containing fixed poses and triangulated points. Closing as answered.
